### PR TITLE
Make languages bookkeeping thread-safe

### PR DESCRIPTION
**Pull request by RandomShaper**
This is my consolidated take resulting from the long discussion in #82578, which this would supersede.

I've needed to make a PR myself to make my changeset clearer and also to base another, urgent, PR on this one.

### CODE REVIEW

File: core/object/worker_thread_pool.cpp [hunk 1]
```diff
@@ -60,6 +61,14 @@ void WorkerThreadPool::_process_task(Task *p_task) {
 		set_current_thread_safe_for_nodes(false);
 		pool_thread_index = thread_ids[Thread::get_caller_id()];
 		ThreadData &curr_thread = threads[pool_thread_index];
+		// Since the WorkerThreadPool is started before the script server,
+		// it's pre-created threads can't have ScriptServer::thread_enter() called on them early.
```
Comment (YuriSizov) [inline core/object/worker_thread_pool.cpp:65]:
```suggestion
		// its pre-created threads can't have ScriptServer::thread_enter() called on them early.
```

I can only say that all this makes sense on the spiritual level, but can't really review it. So here's a grammar nitpick instead :)

**Review by reduz — APPROVED** [review]
seems good to me

**Review by AThousandShips — APPROVED** [review]
LGTM

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**